### PR TITLE
code review

### PR DESCRIPTION
**Pull request by ccoVeille**
- **remove useless receivers**
- **remove useless else statement**
- **avoid wrapping issues**

### CODE REVIEW

**Comment by ccoVeille** [comment]
I started looking at the code and I found bad practice that could lead to panic or error if the code is refactored

**Comment by itchyny** [comment]
This is meaningless change applying unthoughtful _best practices_. The errors are controlled in the package and `errors.Is` is reasonable only when the error comes from another package, also I don't like for performance reasons. Return statement in else is intentional, not to catch unexpected failure of forgetting return in previous branches, possibly caused by nested branches. The first commit about useless receivers is the only one I can merge in the changes.

**Comment by ccoVeille** [comment]
I get your points. I disagree with some of them, but OK, I don't mind.

I'll drop the commits you find useless.

File: cli/inputs.go [hunk 1]
```diff
@@ -89,15 +90,16 @@ func (i *jsonInputIter) Next() (any, bool) {
...
 		}
 		var offset *int64
 		var line *int
-		if err, ok := err.(*json.SyntaxError); ok {
-			err.Offset -= i.offset
-			offset, line = &err.Offset, &i.line
+		var syntaxErr *json.SyntaxError
+		if errors.As(err, &syntaxErr) {
+			syntaxErr.Offset -= i.offset
+			offset, line = &syntaxErr.Offset, &i.line
```
Comment (itchyny) [inline cli/inputs.go:102]: This part is good, the error comes from json package.

File: cli/inputs.go [hunk 2]
```diff
@@ -275,15 +277,16 @@ func (i *streamInputIter) Next() (any, bool) {
...
```
Comment (itchyny) [inline cli/inputs.go:289]: ditto

File: func.go [hunk 3]
```diff
@@ -2065,7 +2068,7 @@ func compileRegexp(re, flags string) (*regexp.Regexp, error) {
 	}
 	r, err := regexp.Compile(re)
 	if err != nil {
-		return nil, fmt.Errorf("invalid regular expression %q: %s", re, err)
+		return nil, fmt.Errorf("invalid regular expression %q: %w", re, err)
```
Comment (itchyny) [inline func.go:2071]: This change is find, but I don't think gojq library users unwrap the execution errors though.

File: module_loader.go [hunk 4]
```diff
@@ -98,7 +99,7 @@ func (l *moduleLoader) LoadJSONWithMeta(name string, meta map[string]any) (any,
 	for {
 		var val any
 		if err := dec.Decode(&val); err != nil {
-			if err == io.EOF {
+			if errors.Is(err, io.EOF) {
```
Comment (itchyny) [inline module_loader.go:102]: good.

**Comment by ccoVeille** [comment]
I will need time and review from you to make sure the place you want me to use errors.Is/errors.As

I'm splitting this PR in two then.

#250 
#251 

I'm dropping the refactoring on switch/else.

**Pull request closed without merging**